lib/neurx.ex

```diff
@@ -3,16 +3,126 @@ defmodule Neurx do
   Documentation for Neurx.
   """
 
-  use Application
+  alias Neurx.{Network}
 
-  def start(_type, _args) do
-    import Supervisor.Spec, warn: true
+  @loss_types ["MSE"]
+  @default_loss %{type: "MSE"}
 
-    children = [
-      # Define children/workers and sub-supervisors to be supervised
-      # worker(Neurx.Worker, [arg1, arg2, arg3]), etc..
-    ]
-    opts = [strategy: :one_for_one, name: Neurx.Supervisor]
-    Supervisor.start_link(children, opts)
+  @activation_types ["Sigmoid", "Relu"]
+  @default_activation "Sigmoid"
+
+  @optim_types ["SGD"]
+  @default_optim %{type: "SGD", learning_rate: 0.1}
+  @default_learning_rate 0.1
+
+  @doc """
+  Builds the network.
+  """
+  def build(config) do
+    if config == %{} or config == nil do
+      raise "Configuration is empty or nil."
+    end
+
+    if config[:input_layer] <= 0 do
+      raise "Invalid size of input layer."
+    end
+
+    # Sanitize the output layer config.
+    output_layer =
+      if config[:output_layer] != nil do
```

Comment:
```
cond do
  config[:output_layer] == nil -> raise "[neurx.build] Output layer unspecified"
  config[:output_layer][:size] == 0 -> raise "[neurx.build] Output layer made 0, it needs to exist!"
  _ -> 
    %{size: config[output_layer][:size],
    activation: sanitize_activation_functions(config[:output_layer][:activation])}
end
```

Reply:
Cond won't handle doing a condition, so if we change if (size <= 0) to size == 0 no exception will be raised for negative numbers. The if statements will catch this.